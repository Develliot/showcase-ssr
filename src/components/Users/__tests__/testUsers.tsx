import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Users } from '../';

const mockUsers = [
    {
        email: 'angela.castro@example.com',
        location: {
            street: {
                number: '8267',
                name: 'Grove Road',
            },
            city: 'Stevenage',
            state: 'Durham',
            postcode: 'BD18 8JD',
            coordinates: {
                latitude: 61.2773,
                longitude: 45.9598,
            },
        },
        login: {
            uuid: '1',
        },
        name: { first: 'Angela', last: 'Castro' },
        picture: {
            large: 'https://randomuser.me/api/portraits/women/45.jpg',
        },
    },
];

const mockRetry = jest.fn();

const mockProps = {
    isError: false,
    isLoading: false,
    users: mockUsers,
    retry: mockRetry,
};

it('renders with users', () => {
    const wrapper: ShallowWrapper = shallow(<Users {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
});

it('renders with loading', () => {
    const wrapper: ShallowWrapper = shallow(
        <Users {...mockProps} isLoading={true} />
    );
    expect(wrapper).toMatchSnapshot();
});

it('renders with error', () => {
    const wrapper: ShallowWrapper = shallow(
        <Users {...mockProps} isError={true} />
    );
    expect(wrapper).toMatchSnapshot();
});
