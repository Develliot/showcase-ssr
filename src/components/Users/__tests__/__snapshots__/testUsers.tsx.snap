// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with error 1`] = `
<Fragment>
  <ErrorMessage
    errorMessage="Unable to retrieve user data"
  />
</Fragment>
`;

exports[`renders with loading 1`] = `
<Fragment>
  <Carousel>
    <UserCardLoading />
    <UserCardLoading />
    <UserCardLoading />
  </Carousel>
</Fragment>
`;

exports[`renders with users 1`] = `
<Fragment>
  <Carousel
    onPositionChanged={[Function]}
  >
    <UserCard
      color="#A93F55"
      key="1"
      user={
        Object {
          "email": "angela.castro@example.com",
          "location": Object {
            "city": "Stevenage",
            "coordinates": Object {
              "latitude": 61.2773,
              "longitude": 45.9598,
            },
            "postcode": "BD18 8JD",
            "state": "Durham",
            "street": Object {
              "name": "Grove Road",
              "number": "8267",
            },
          },
          "login": Object {
            "uuid": "1",
          },
          "name": Object {
            "first": "Angela",
            "last": "Castro",
          },
          "picture": Object {
            "large": "https://randomuser.me/api/portraits/women/45.jpg",
          },
        }
      }
    />
  </Carousel>
</Fragment>
`;
